src/pykeen/evaluation/evaluator.py

```diff
@@ -555,9 +557,9 @@ def evaluate(
 
     # Prepare for result filtering
     if filtering_necessary or positive_masks_required:
-        if training_mapped_triples is None:
-            raise ValueError
-        all_pos_triples = torch.cat([training_mapped_triples, mapped_triples], dim=0)
+        if additional_filtered_triples is None:
```

Comment:
Shouldn't we also allow filtered evaluation with the evaluation triples alone?

Reply:
maybe? but changing functionality isn't the job of this PR ;)

Reply:
This is not the standard use-case for usage from the training pipeline though, so maybe add a warning.

Reply:
You think it should be the following? Otherwise I'm not really sure what you want
```
if filtering_necessary or positive_masks_required:
        if additional_filtered_triples is None:
            all_pos_triples = mapped_triples
        else:
            all_pos_triples = torch.cat([additional_filtered_triples, mapped_triples], dim=0)
        all_pos_triples = all_pos_triples.to(device=device)
    else:
        all_pos_triples = None
```

Reply:
Previously, the model brought the training triples factory along for the ride, so we didn't have to care if it was available or not. Now, since we're passing it as `additional_filtered_triples`, we only need to do so if filtering is going to happen. Since this is optional, this final check and raise ValueError is if you passed an invalid combination of parameters. It still works the same way through the pipeline

Reply:
> You think it should be the following? Otherwise I'm not really sure what you want

Yes, exactly this.

Reply:
@lvermue @mberr we need to reinvestigate this. Should we raise an error? Or should we take away the default value of None for additional_filtered_triples so a type error is thrown if the usage isn't updated?